ProjectTests fails on some seed

### CI Link

local

### Repro line

./gradlew ':x-pack:plugin:esql:compute:test' --tests "org.elasticsearch.compute.operator.ProjectOperatorTests.testProjection" -Dtests.seed=91943554DA6E3559 -Dtests.locale=ar-SA -Dtests.timezone=Indian/Comoro -Druntime.java=21

### Does it reproduce?

Yes

### Applicable branches

main

### Failure history

_No response_

### Failure excerpt

```
  2> java.lang.AssertionError: java.lang.IllegalStateException: can't release already released block [IntVectorBlock[vector=ConstantIntVector[positions=3, value=0]]]
        at __randomizedtesting.SeedInfo.seed([91943554DA6E3559:69BE85F199AA9B5]:0)
        at org.elasticsearch.core.Releasables.closeExpectNoException(Releasables.java:49)
        at org.elasticsearch.compute.data.Page.releaseBlocks(Page.java:224)
        at org.elasticsearch.compute.operator.ProjectOperatorTests.testProjection(ProjectOperatorTests.java:79)

        Caused by:
        java.lang.IllegalStateException: can't release already released block [IntVectorBlock[vector=ConstantIntVector[positions=3, value=0]]]
            at org.elasticsearch.compute.data.IntVectorBlock.close(IntVectorBlock.java:76)
            at org.elasticsearch.core.IOUtils.close(IOUtils.java:71)
            at org.elasticsearch.core.IOUtils.close(IOUtils.java:87)
            at org.elasticsearch.core.IOUtils.close(IOUtils.java:63)
            at org.elasticsearch.core.Releasables.close(Releasables.java:73)
            at org.elasticsearch.core.Releasables.close(Releasables.java:41)
            at org.elasticsearch.core.Releasables.closeExpectNoException(Releasables.java:47)
            ... 2 more
```

I'm unsure if the test is generating sensible projection configurations. This may be an invalid thing.

--- a/x-pack/plugin/esql/compute/src/test/java/org/elasticsearch/compute/operator/ProjectOperatorTests.java
+++ b/x-pack/plugin/esql/compute/src/test/java/org/elasticsearch/compute/operator/ProjectOperatorTests.java
@@ -17,13 +17,16 @@ import org.elasticsearch.compute.data.BlockFactory;
 import org.elasticsearch.compute.data.IntBlock;
 import org.elasticsearch.compute.data.LongBlock;
 import org.elasticsearch.compute.data.Page;
+import org.elasticsearch.core.Releasables;
 import org.elasticsearch.core.Tuple;
 import org.elasticsearch.indices.breaker.CircuitBreakerService;
 import org.junit.After;
 import org.junit.Before;
 
 import java.util.Arrays;
+import java.util.HashSet;
 import java.util.List;
+import java.util.Set;
 import java.util.stream.LongStream;
 
 import static org.hamcrest.Matchers.equalTo;
@@ -71,12 +74,15 @@ public class ProjectOperatorTests extends OperatorTestCase {
         var out = projection.getOutput();
         assertThat(randomProjection.size(), lessThanOrEqualTo(out.getBlockCount()));
 
+        Set<Block> blks = new HashSet<>();
         for (int i = 0; i < out.getBlockCount(); i++) {
             var block = out.<IntBlock>getBlock(i);
             assertEquals(block, page.getBlock(randomProjection.get(i)));
+            blks.add(block);
         }
+
         // close all blocks separately since the same block can be used by multiple columns (aliased)
-        out.releaseBlocks();
+        Releasables.closeWhileHandlingException(blks.toArray(new Block[0]));
     }
 
     private List<Integer> randomProjection(int size) {

